### PR TITLE
Hubspot form integration

### PR DESCRIPTION
**Pull request by Shruti-Apte**
### **Notion :**
https://www.notion.so/saeloun/Integration-with-Marketing-tool-Hubspot-4836e0626b8843378996ea943ccd1d13?pvs=4
### **What :**
- Hubspot form integration
### **Why :**
- To keep a track of user signups on miru.
### **Preview :**
Screenshot 2023-07-31 at 4 14 08 PM
### **Todos** (for testing):

Won't send a req with already existing email Ids as we are sending the hubspot req after the signup req.

### CODE REVIEW

**Review by sanjibroy360 — APPROVED** [review]

File: app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx [hunk 1]
```diff
@@ -39,6 +39,9 @@ const SignUpForm = () => {
       const { first_name, last_name, email, password, confirm_password } =
         values;
 
+      const portalId = 23903467;
+      const formId = "3266596a-b69b-40a2-9cd4-58ef522a270a";
```
Comment (akhilgkrishnan) [inline app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx:43]: Can we fetch these values from env?

File: app/javascript/src/apis/authentication.ts [hunk 2]
```diff
@@ -34,13 +34,20 @@ const googleAuth = () =>
     })
     .get("users/auth/google_oauth2");
 
+const hubspot = async (portalId, formId, payload) =>
+  await axios.post(
+    `https://api.hsforms.com/submissions/v3/integration/submit/${portalId}/${formId}`,
```
Comment (akhilgkrishnan) [inline app/javascript/src/apis/authentication.ts:39]: We should use env's for external references

**Review by akhilgkrishnan — COMMENTED** [review]
@Shruti-Apte Added some minor suggestions.

File: app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx [hunk 3]
```diff
@@ -47,6 +50,25 @@ const SignUpForm = () => {
...
+          {
+            name: "firstname",
+            value: first_name,
+          },
+          {
+            name: "lastname",
+            value: last_name,
+          },
+        ],
+      });
```
Comment (akhilgkrishnan) [inline app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx:70]: Can't we use backend for making this request?

Comment (Shruti-Apte) [reply app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx:70]: I'll discuss this with supriya.

**Comment by prasanthchaduvula** [comment]
Closes this PR as we are handling the integration from backend #1444

**Pull request closed without merging**